app/src/main/java/org/mozilla/focus/utils/UrlUtils.java

```diff
@@ -82,38 +84,32 @@ public static String getSearchTermsOrUrl(Context context, String url) {
     }
 
     /**
-     * @return If a url matches the form of a first search URL for the current default search engine.
+     * @return The detected search engine or null if one was not matched with the URL
      */
-    private static boolean isDefaultSearchUrl(Context context, String urlSearch) {
+    private static SearchEngine isDefaultSearchUrl(String urlSearch) {
         if (TextUtils.isEmpty(urlSearch)) {
-            return false;
+            return null;
         }
 
-        final SearchEngine searchEngine = SearchEngineManager.getInstance()
-                .getDefaultSearchEngine(context);
-
-        final String baseSearchUrl = searchEngine.getBaseSearchUrl();
-        if (TextUtils.isEmpty(baseSearchUrl)) {
-            return false;
-        }
-        // %7B is a placeholder for search terms - if it is a default search engine url, the urls should match before the placeholder.
-        if (baseSearchUrl.contains("%7B")) {
-            final int searchPlaceholderIndex = baseSearchUrl.lastIndexOf("%7B");
-            return urlSearch.startsWith(baseSearchUrl.substring(0, searchPlaceholderIndex));
-        } else {
-            return false;
+        final String representativeSnippet = UrlUtils.getRepresentativeSnippet(urlSearch);
+        final List<SearchEngine> searchEngines = SearchEngineManager.getInstance().getSearchEngines();
+        for (SearchEngine searchEngine : searchEngines) {
+            if (representativeSnippet.contains(searchEngine.getName().toLowerCase())) {
+                return searchEngine;
+            }
```

Comment:
I wonder why we iterate over all search engines now?

Also I'm a bit afraid that checking whether the name of the search engine is in the URL is a bit fragile. For example the name for the German Wikipedia search engine is "Wikipedia (de)".